src/contracts/libraries/OwnershipManagerLib.sol

```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "../interfaces/IOwnershipManager.sol";
+
+
+library OwnershipManagerLib {
+    function getIdentifier(address agent, address ownershipManager) internal view returns (bytes20) {
+        return IOwnershipManager(ownershipManager).getIdentifier(agent, address(this), bytes4(msg.data[0:4]));
```

Comment:
passing selector > msg.sig > slicing the calldata+casting

Reply:
passing a storage reference to the mapping that getIdentifier uses would probably be cleaner here